documentation/src/main/asciidoc/user_guide/eviction.adoc

```diff
@@ -170,18 +172,45 @@ stay in the cache or _maximum idle time_ , time it's allowed to be untouched
 
 .  _Expiration_ is a top-level construct, represented in the configuration as well as in the cache API.
 .  While eviction is _local to each cache instance_ , expiration is _cluster-wide_ .  Expiration lifespans and maxIdle values are replicated along with the cache entry.
-.  While maxIdle is replicated, expiration due to maxIdle is not cluster wide, only lifespan.  As such it is not recommended to use maxIdle in a clustered cache.
+.  Max idle requires additional network messages to work in a cluster.  As such using maxIdle in a clustered cache can cause some operations to be slower at times.
 .  Expiration lifespan and maxIdle are also persisted in CacheStores, so this information survives eviction/passivation.
 
+
+==== Max Idle Details
+
+Max idle expiration occurs when an entry has not been accessed within a given time frame. This poses a more interesting problem than lifespan, which is very simple. Thus this type of expiration behaves slightly different in a local versus cluster cache.
+
+===== Local Max Idle
+
+Max idle is pretty straightforward in a non clustered cache. It removes expired entries when accessed directly (ie. Cache.get), also removes expired entries entries when iterated upon (ie. Cache.size) or via expiration reaper.
+
+===== Cluster Max Idle
+
+Max idle in a cluster is quite a bit more complicated. Since each node can have a different access time, an entry is not removed until all owners of said entry say it hasn't been accessed in the given amount of time.
+
+This check is done when a given node finds an entry has expired via maxIdle. It then will ask all the other owners what their last access time was (if it hasn't expired). If all owners say that the entry has expired, then it will not be returned to the user and removed based on the operation.
+
+WARN: Clustered caches should always enable reaper when using max idle expiration due to some limitations listed in the section below.
+
+When using max idle, it is important to keep the following points in mind:
+
+. A Cache.get in a clustered cache will return the "expired" entry if at least one owner finds that this entry is not expired. All owners will then update their last access time to the highest number, essentially refreshing this entry until that time.
```

Comment:
I think I may have insisted too much on using the max timestamp from the purge thread, but I was thinking that a regular read would still update the timestamp to the current time.

Reply:
Sorry the wording may have confused you, this is a regular read that found an expired entry. I will clear it up.

Reply:
I know it's about a regular read, I'm just confused about "All owners will then update their last access time to the highest number, essentially refreshing this entry until that time." Why use the highest number and not the current time?

Reply:
Yes, I agree. Sorry I haven't tuned my brain for expiration totally. The highest number is only for the expiration reaper. Let me confirm how this behaving again.

Reply:
I believe my intent was to differentiate them, but for now we only update to the latest timestamp, since the behavior can be shared between. This keeps correct behavior for reaper and may slow down gets (depending on usage). I can log a JIRA to add these improvements later if need be. I believe due to timing I didn't concentrate on this and the other improvement mentioned in the prior comment.

Reply:
Actually after I slept on this, this is wrong. We _have_ to update the timestamp on a get. Otherwise the entry may expire preemptively again. I will see how much work it is to get this done.